pkg/pattern/drain/chunk.go

```diff
@@ -51,7 +51,7 @@ func (c Chunk) ForRange(start, end, step model.Time) []logproto.PatternSample {
 	}
 	first := c.Samples[0].Timestamp
 	last := c.Samples[len(c.Samples)-1].Timestamp
-	if start >= end || first >= end || last < start {
+	if last < first || start >= end || first >= end || last < start {
```

Comment:
this might not cover all the cases since the in-between samples could be out-of-order?

Reply:
Good point - I'll add some more test cases